### ISSUE TITLE
Non-interactive content loses virtual focus when ‘activated’ in JAWS

### ISSUE THREAD

**Issue by aardrian** [FreedomScientific/VFO-standards-support#701]
### Summary

When navigating by content (headings, arrowing through the page, etc), the first time a user press `Enter` or `Space` while the virtual cursor is on non-interactive content after a fresh page load (but not a refresh), JAWS will move virtual cursor focus to the top of the page (rather, it loses focus and the user has to start over). 

To re-create:

1. [Open this pen in Debug mode](https://cdpn.io/aardrian/debug/ExemKjE) in a new window using Chrome/JAWS;
2. Press `Tab` to put focus on the Nu HTML Checker link;
3. Press `H` to navigate to any heading (I suggest starting with the bottom ones) or press `↓` to move further into the paragraph or any other button to move the virtual cursor to plain content;
4. Press `Enter` or `Space` while the virtual cursor is on the content;
5. Observe virtual cursor focus is lost and subsequent navigation starts from the top of the page.

### Expected result

Virtual cursor does not move.

### Actual result

Virtual cursor focus/position is lost and subsequent navigation starts from the top of the page. Received some [confirmation on Mastodon](https://hachyderm.io/@tristan/109956163201598066) as well.

### Example

* https://cdpn.io/aardrian/debug/ExemKjE (debug mode)
* https://codepen.io/aardrian/pen/ExemKjE (editor view)

#### JAWS version and build number

* JAWS 2023.2212.23
 
#### Operating System and version

* Windows 11

#### Browser and version:

* Chrome 110
* Edge 110

**Comment by tmthywynn8** [comment]
Can also confirm this in Firefox Nightly 112.0a1 as well with [Accessibility cache checked](https://blog.mozilla.org/accessibility/significant-improvements-for-screen-readers-now-in-nightly-firefox/) with JAWS Version 2023.2212.23 in Windows 10 22H2.

**Comment by JAWS-test** [comment]
I can confirm the problem.

@aardrian Why would someone try to activate a non-operating element with ENTER? I can only imagine that ENTER is to activate the link where the focus is still (also visible), but I suspect that no one would try this once the virtual cursor was moved away from the link

**Comment by aardrian** [comment]
> Why would someone try to activate a non-operating element with ENTER?

1. In the case of a sighted screen reader user, if the virtual cursor visual indicator leads the user to believe the wrong thing has focus;
2. Accidentally;
3. A tester unfamiliar with screen readers tries it (and files a defect against a component that is really a JAWS bug as a result).

**Comment by tmthywynn8** [comment]
I'm not an accessibility tester by trade so probably don't have much say in this (feel free to mark it as off-topic), but as an end user, I sometimes press *Enter* on non-interactive content, e.g., to attempt to expand an  answer in an FAQ. If I get thrown back to the top of the page for pressing *Enter* on the wrong bit of text, I would then be trying to get around two issues, one from the web developer and one from screen reader implementation.

**Comment by JAWS-test** [comment]
@tmthywynn8 

> I sometimes press Enter on non-interactive content, e.g., to attempt to expand an answer in an FAQ

Non-interactive content is non-interactive i.e. nothing can be expanded there. In FAQ, if the answer can be shown and hidden by activating the question, then the question is interactive content (e.g. summary element within details element or button element with ARIA attribute aria-expanded). So in this case the problem would not occur

**Comment by aardrian** [comment]
@JAWS-test The use case @tmthywynn8 outlines is likely accounting for when a developer puts a click handler on a heading without otherwise making it into a proper control. I am guessing they have encountered these terrible developer practices and learned that sometimes it is worth trying to activate if the context suggests there may be more hidden.

I amended my example (https://cdpn.io/aardrian/debug/ExemKjE) to add a heading level 3 at the end ("Just a Normal HTML Heading") with a click handler. If you navigate to the heading and hit `Enter` or `Space`, you will get a browser alert. Then you can also experience the loss of virtual focus.

This valid use case is why I amended my numbered list above to add it.

**Comment by stevefaulkner** [comment]
@aardrian  I could not reproduce this issue, am probably missing something. Ping me when you have time and will go through it with you on a call?

**Comment by bflon8** [comment]
I was also able to replicate this issue. Anyone have a fix or is there any movement here?

**Comment by aardrian** [comment]
Still an issue in JAWS 2024.

**Comment by joelearn** [comment]
I found that when a disabled button is off-screen, pressing the Down key until the button is announced, then pressing Enter, causes the focus to be lost - not to the top of the page, but completely outside of the browser. Although this might happen only by accident, the expectation is that nothing should occur in such a case.
